pkg/serializer/internal/metrics/events.go

```diff
@@ -67,7 +67,7 @@ func (events Events) Marshal() ([]byte, error) {
 }
 
 func (events Events) getEventsBySourceType() map[string][]*event.Event {
-	eventsBySourceType := make(map[string][]*event.Event)
+	eventsBySourceType := make(map[string][]*event.Event, len(events))
```

Comment:
Would it make sense to change this to the (expected) number of unique source types, rather than total number of events? It seems source types are rather limited in number, so this may degrade memory usage if a significant number of events are processed.